src/utilities.js

```diff
@@ -0,0 +1,113 @@
+import * as Constants from './constants';
+
+export function addLotToPortfolio(ticker) {
+  let portfolio = JSON.parse(window.localStorage.getItem('portfolio'));
+  if (!portfolio) {
+    window.localStorage.setItem(
+      'portfolio',
+      JSON.stringify({
+        name: 'To The Moon',
+        lots: [],
+      })
+    );
+    portfolio = JSON.parse(window.localStorage.getItem('portfolio'));
+  }
+  portfolio.lots.push({
+    symbol: ticker.toUpperCase(),
+    buyShares: 0,
+    buyDate: null,
+    broker: null,
+    buyPrice: null,
+    sellShares: 0,
+    sellDate: null,
+    sellPrice: null,
+  });
+  window.localStorage.setItem('portfolio', JSON.stringify(portfolio));
+}
+
+// TODO: need to convert this to sum values
+export const getDataForChart = (ticker) => {
+  const pricesJson = Constants.API_PRICES[ticker];
+  const xAxisLabels = [];
+  const yAxisLabels = [];
+  for (let i = 0; i < pricesJson.length; i++) {
+    xAxisLabels.push(pricesJson[i].date.split('T')[0]);
+    yAxisLabels.push(pricesJson[i].close);
+  }
+
+  const metaData = Constants.API_META[ticker];
+  const name = metaData['name'];
+
+  return { xAxisLabels, yAxisLabels, name };
+};
+
+export function formatDate(date) {
+  var d = new Date(date),
+    month = '' + (d.getMonth() + 1),
```

Comment:
month = ('0' + d.getMonth() + 1).substr(-2)

Reply:
huh? why? this doesn't seem to work for double-digit months

Reply:
Correction:

`month = ('0' + (d.getMonth() + 1)).substr(-2)

Reply:
I'm still not seeing how this is better than what I already have?

Reply:
Yours requires an additional `if (month.length < 2) month = '0' + month;` (and similar for `day`) ... my code is shorter and probably about as efficient.

Another possibility:
`month = (d.getMonth() < 10 ? '0' : '') + d.getMonth();`

As this seems to be a fairly common operation, you might want to encapsulate it in a utility function:
`const zeroPadTo2 = (d) => (d < 10 ? '0' : '') + d;

Reply:
Or, even better (more general for very little extra effort):
`// Zero pad i on the left to n characters`
`const zeroPad = (i, n) => ('0000000000' + i).substr(-n);`

Reply:
An improvement:

`const lpad = (str, fill, width) => fill.repeat(Math.max(0, width-str.length)) + str;`
`const zeroPad = (num, width) => lpad(num.toString(), "0", width);`

Another possibility for `lpad` is:
`const lpad = (str, fill, width) => (fill.repeat(width) + str).substr(-width);`

Reply:
done